packages/flutter_tools/gradle/module_plugin_loader.gradle

```diff
@@ -5,41 +5,26 @@
 // This file is included from `<module>/.android/include_flutter.groovy`,
 // so it can be versioned with the Flutter SDK.
 
-import groovy.json.JsonSlurper
+import java.nio.file.Paths
+
+File pathToThisDirectory = buildscript.sourceFile.parentFile
+apply from: Paths.get(pathToThisDirectory.absolutePath, "src", "main", "groovy", "shared.groovy")
 
 def moduleProjectRoot = project(':flutter').projectDir.parentFile.parentFile
 
-def object = null;
-String flutterModulePath = project(':flutter').projectDir.parentFile.getAbsolutePath()
-// If this logic is changed, also change the logic in app_plugin_loader.gradle.
-def pluginsFile = new File(moduleProjectRoot, '.flutter-plugins-dependencies')
-if (pluginsFile.exists()) {
-    object = new JsonSlurper().parseText(pluginsFile.text)
-    assert object instanceof Map
-    assert object.plugins instanceof Map
-    assert object.plugins.android instanceof List
-    // Includes the Flutter plugins that support the Android platform.
-    object.plugins.android.each { androidPlugin ->
-        assert androidPlugin.name instanceof String
-        assert androidPlugin.path instanceof String
-        // Skip plugins that have no native build (such as a Dart-only
-        // implementation of a federated plugin).
-        def needsBuild = androidPlugin.containsKey('native_build') ? androidPlugin['native_build'] : true
-        if (!needsBuild) {
-            return
-        }
-        def pluginDirectory = new File(androidPlugin.path, 'android')
-        assert pluginDirectory.exists()
-        include ":${androidPlugin.name}"
-        project(":${androidPlugin.name}").projectDir = pluginDirectory
-    }
+List<Map<String, Object>> nativePlugins = nativePluginLoader.getPlugins(moduleProjectRoot)
```

Comment:
Can you leave a comment as to why this has to Object and not a more precise type?

Reply:
I moved the description not to `nativePluginLoader` to why this needs to be an Object 4612b4b. See also https://github.com/flutter/flutter/pull/137115#discussion_r1390456444
The `dependencies` property is also used in `configurePluginDependencies`, therefore it can be either a `String` or a `List<String>` or a `bool`.

Reply:
Done in 7c4a67dfb40c03d4f07fd24749cf465d53a1fd70